pkg/remote/proxy.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"context"
+
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/googleapis/api/httpbody"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// Proxy implements a local proxy for a remote Registry server.
+// This allows tests written to the generated "RegistryServer" and "AdminServer" interfaces to be run against remote servers.
```

Comment:
Is this just for testing? It almost seems like it could  be general-purpose (aside from the Open() function behavior). If this _is_ just intended for supporting tests, we should probably place it in a package under the "tests" folder or some other similar "test" oriented folder.

Reply:
I just moved the remote proxy into server/registry/test/remote